### ISSUE TITLE
JPG embedded ICC profile causing problems

### ISSUE THREAD

**Issue by ChrisPinewood** [lovell/sharp#521]
I'm having trouble with the following JPG image when trying to convert it to PNG:

[Problem image](https://drive.google.com/file/d/0B7HuGLc19qSoMnQ2WGVVS29hbTQ/view?usp=sharing)

It appears to be trying to use the embedded CMYK ICC profile but failing, the image itself is marked as sRGB.  I did wonder if just stripping out the embedded ICC profile out would work, but I'm not sure how to do this, or whether I should be taking a different approach.

This is the code I'm using to convert (as well as resize):
```
  sharp(filename)
    .resize(requiredSize.width, requiredSize.height)
    .background({r: 0, g: 0, b: 0, a: 0})
    .embed()
    .png()
    .toBuffer().then((outputBuffer) => {
      resolve(outputBuffer);
    })
    .catch((err) => {
      console.log('Error creating image:', err);
      reject(err);
    });
```

It's very possible I'm doing something very silly in the above!

Many thanks in advance.

**Comment by lovell** [comment]
@ChrisPinewood Your pinewood-solutions website appears to be compromised.

```html
<!--dd4185--><script type="text/javascript" src="http://firma-krajewscy ... "></script><!--/dd4185-->
```

**Comment by ChrisPinewood** [comment]
Thank you Lovell - I'll sort that!

Any reason for the problem that I'm seeing yet?

**Comment by lovell** [comment]
Looks like this module has never removed ICC profiles from PNG output; nice bug.

libvips' PNG saver doesn't use `strip` like all the others, so the profile has to be removed instead. Commit 032bb7e adds a test and the fix.

**Comment by ChrisPinewood** [comment]
Fantastic - thank you Lovell for such a quick response!

**Comment by rodrigoalviani** [comment]
I have the same problem, but without convert to PNG. Just when crop a image. Always jpg 2 jpg.
`vips warning: icc_transform: embedded profile incompatible with image`

**Comment by lovell** [comment]
@rodrigoalviani `icc_transform: embedded profile incompatible with image` usually indicates a CMYK profile in a RGB image (or vice versa). Set the `VIPS_WARNING` env variable to silence warnings from libvips.

**Comment by lovell** [comment]
This is fixed in v0.16.0, now available via npm.
